packages/cli/test/unit/commands/promote.test.ts

```diff
@@ -0,0 +1,380 @@
+  });
+
+  it('should error if promote request fails', async () => {
+    const { cwd, previousDeployment } = initPromoteTest({
+      promotePollCount: 10,
+      promoteStatusCode: 500,
+    });
+
+    client.setArgv('promote', previousDeployment.id, '--yes', '--cwd', cwd);
+    const exitCodePromise = promote(client);
+
+    await expect(client.stderr).toOutput(
+      `Fetching deployment "${previousDeployment.id}" in ${previousDeployment.creator?.username}`
+    );
+
+    // we need to wait a super long time because fetch will return on 500
+    await expect(client.stderr).toOutput('Response Error (500)', 20000);
+
+    await expect(exitCodePromise).resolves.toEqual(1);
+  });
+
+  it('should error if rollback fails (no aliases)', async () => {
+    const { cwd, previousDeployment } = initPromoteTest({
+      promoteJobStatus: 'failed',
+    });
+    client.setArgv('promote', previousDeployment.id, '--yes', '--cwd', cwd);
+    const exitCodePromise = promote(client);
+
+    await expect(client.stderr).toOutput(
+      `Fetching deployment "${previousDeployment.id}" in ${previousDeployment.creator?.username}`
+    );
+    await expect(client.stderr).toOutput('Promote in progress');
+    await expect(client.stderr).toOutput(
+      `Error: Failed to remap all aliases to the requested deployment ${previousDeployment.url} (${previousDeployment.id})`
+    );
+
+    await expect(exitCodePromise).resolves.toEqual(1);
+  });
+
+  it('should error if promote fails (with aliases)', async () => {
+    const { cwd, previousDeployment } = initPromoteTest({
+      promoteAliases: [
+        {
+          alias: { alias: 'foo', deploymentId: 'foo_123' },
+          status: 'completed',
+        },
+        {
+          alias: { alias: 'bar', deploymentId: 'bar_123' },
+          status: 'failed',
+        },
+      ],
+      promoteJobStatus: 'failed',
+    });
+    client.setArgv('promote', previousDeployment.id, '--yes', '--cwd', cwd);
+    const exitCodePromise = promote(client);
+
+    await expect(client.stderr).toOutput(
+      `Fetching deployment "${previousDeployment.id}" in ${previousDeployment.creator?.username}`
+    );
+    await expect(client.stderr).toOutput('Promote in progress');
+    await expect(client.stderr).toOutput(
+      `Error: Failed to remap all aliases to the requested deployment ${previousDeployment.url} (${previousDeployment.id})`
+    );
+    await expect(client.stderr).toOutput(
+      `  ${chalk.green('completed')}    foo (foo_123)`
+    );
+    await expect(client.stderr).toOutput(
+      `  ${chalk.red('failed')}       bar (bar_123)`
+    );
+
+    await expect(exitCodePromise).resolves.toEqual(1);
+  });
+
+  it('should error if deployment times out', async () => {
+    const { cwd, previousDeployment } = initPromoteTest({
+      promotePollCount: 10,
+    });
+    client.setArgv(
+      'promote',
+      previousDeployment.id,
+      '--yes',
+      '--cwd',
+      cwd,
+      '--timeout',
+      '2s'
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (non-blocking):** ‏make timeout smaller

Can we reduce the timeout to something trivially small to reduce the amount of fixed waiting time the tests have?

Reply:
I lowered it from 2s to 1s and then down to 1ms.